### PR TITLE
Unified precision

### PR DESCRIPTION
**Pull request by Ashuaidehao**
ApplicationLog and RpcNep17Tracker will return BigInteger via RPC.
Relate to  https://github.com/neo-project/neo-modules/pull/496

### CODE REVIEW

File: src/ApplicationLogs/LogReader.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ public static JObject TxLogToJson(Blockchain.ApplicationExecuted appExec)
             trigger["trigger"] = appExec.Trigger;
             trigger["vmstate"] = appExec.VMState;
             trigger["exception"] = GetExceptionMessage(appExec.Exception);
-            trigger["gasconsumed"] = new BigDecimal(new BigInteger(appExec.GasConsumed), NativeContract.GAS.Decimals).ToString();
+            trigger["gasconsumed"] = new BigInteger(appExec.GasConsumed).ToString();
```
Comment (erikzhang) [inline src/ApplicationLogs/LogReader.cs:71]:
```suggestion
            trigger["gasconsumed"] = appExec.GasConsumed.ToString();
```

Comment (erikzhang) [reply src/ApplicationLogs/LogReader.cs]: And other places.

**Comment by bettybao1209** [comment]
Is it ok that `GetWalletBalance` in https://github.com/neo-project/neo-modules/blob/master/src/RpcServer/RpcServer.Wallet.cs#L78 return decimal?  Since `GetAvailable` in neo core returns decimal.

**Review by shargon — APPROVED** [review]

**Comment by shargon** [comment]
> Is it ok that GetWalletBalance in https://github.com/neo-project/neo-modules/blob/master/src/RpcServer/RpcServer.Wallet.cs#L78 return decimal? Since GetAvailable in neo core returns decimal.

All equal, it should be long

**Comment by chenzhitong** [comment]
**GetWalletBalance** 

Iterate over all assets and request the decimal method for each asset? May be very impactful to performance.

**Comment by chenzhitong** [comment]
RpcServer.Wallet.cs Line 83 
json["balance"] = wallet.GetAvailable(asset_id).Value.ToString();

**Review by shargon — APPROVED** [review]

**Comment by ProDog** [comment]
Tested OK.

**Review by ProDog — APPROVED** [review]

**Comment by cloud8little** [comment]
Ready to merge. @superboyiii

**Comment by roman-khimov** [comment]
This effectively reverts #485 and parts of #378 and makes RPC interface a little less friendly (one needs to get/store proper decimals to correctly process these values).

**Review by chenzhitong — APPROVED** [review]

File: src/RpcClient/Models/RpcApplicationLog.cs [hunk 2]
```diff
@@ -69,7 +67,7 @@ public static Execution FromJson(JObject json)
             {
                 Trigger = json["trigger"].TryGetEnum<TriggerType>(),
                 VMState = json["vmstate"].TryGetEnum<VMState>(),
-                GasConsumed = (long)BigDecimal.Parse(json["gasconsumed"].AsString(), NativeContract.GAS.Decimals).Value,
+                GasConsumed = (long)json["gasconsumed"].AsNumber(),
```
Comment (erikzhang) [inline src/RpcClient/Models/RpcApplicationLog.cs:70]:
```suggestion
                GasConsumed = long.Parse(json["gasconsumed"].AsString()),
```

**Comment by cloud8little** [comment]
> This effectively reverts #485 and parts of #378 and makes RPC interface a little less friendly (one needs to get/store proper decimals to correctly process these values).

since the rpc function is a generic function, and I think it is reasonable to give the control of converting the data to the upper layer applcation (include sdk).

**Review by erikzhang — APPROVED** [review]

**Pull request merged**